fondant/data_io.py

```diff
@@ -134,7 +132,7 @@ def write_index(self, df: dd.DataFrame):
         index_df = df[index_columns]
 
         # set index
-        index_df.set_index("id")
+        index_df = index_df.set_index("id")
```

Comment:
Getting 
```
 "Data has no column 'id': use any column of ['source']"
```
when trying to run `simple_pipeline`. 

Could you try replicating the issue: 
inside `examples/pipelines/simple_pipeline/components/load_from_hub/src`

you can run 
```
python3 main.py  --output_manifest_path a.txt --metadata '{"run_id":"test", "base_path":"soy-audio-379412-kfp-output"}' --dataset_name lambdalabs/pokemon-blip-captions
```